ghcide/src/Development/IDE/Plugin/TypeLenses.hs

```diff
@@ -90,15 +87,24 @@ descriptor plId =
     { pluginHandlers = mkPluginHandler STextDocumentCodeLens codeLensProvider
     , pluginCommands = [PluginCommand (CommandId typeLensCommandId) "adds a signature" commandHandler]
     , pluginRules = rules
+    , pluginCustomConfig = mkCustomConfig properties
     }
 
+properties :: Properties '[PropertyKey "mode" 'TEnum]
```

Comment:
Using the properties is optional for plugins and tactic plugin has its own custom config. 
Could we add it now to signal using `Properties` is the right way?

Reply:
Let's ask review from @isovector